### ISSUE TITLE
Assess and potentially refactor `label_emotion_concepts`

### ISSUE THREAD

**Issue by mclevey** [mclevey/podlm#26]
@tcrick , could you assess the core function I developed for the `label_emotion_concepts` task and consider whether it needs to be refactored? It does what it is supposed to do, but is probably very inefficient. At the moment it is using a pipeline developed by the authors, but we would have more control over the data structures if we just wrote our own. What I've done is a step in that direction but could be better, I think.

**Comment by mclevey** [comment]
Added label

**Comment by tcrick** [comment]
Yep, short and straightforward function. Easy to see at a glance all of the opportunities for optimization, which I'll do shortly after we have a set of test data to run through. 

My guess is that the biggest performance improvement will end up being the transformer model inference, which looks like is done 1 document at a time in the for-loop.

**Comment by tcrick** [comment]
The transformer component of this issue falls under the umbrella of https://github.com/mclevey/podlm/issues/21
